engine/renderer/private/particles/particle_interface.cpp

```diff
@@ -30,6 +28,55 @@ ParticleInterface::ParticleInterface(const std::shared_ptr<ECS>& ecs)
     }
 }
 
+ParticleInterface::~ParticleInterface()
+{
+    for (auto i : _emitterImages)
+    {
+        _context->Resources()->ImageResourceManager().Destroy(i);
+    }
+}
+
+void ParticleInterface::LoadEmitterPresets()
+{
+    // TODO: serialize emitter presets and load from file
+
+    std::shared_ptr<GraphicsResources> resources = { _context->Resources() };
+
+    // hardcoded test emitter preset for now
+    Emitter emitter;
+    emitter.position = glm::vec3(1.0f, 2.0f, 3.0f);
+    emitter.count = 5;
+    emitter.velocity = glm::vec3(1.0f, 5.0f, 1.0f);
+    emitter.mass = 2.0f;
+    emitter.rotationVelocity = glm::vec3(1.0f);
+    emitter.maxLife = 5.0f;
+    emitter.materialIndex = LoadEmitterImage("assets/textures/nogameplay.png");
+    emitter.size = glm::vec2(resources->ImageResourceManager().Access(_emitterImages[0])->width, resources->ImageResourceManager().Access(_emitterImages[0])->height) / static_cast<float>(glm::max(resources->ImageResourceManager().Access(_emitterImages[0])->width, resources->ImageResourceManager().Access(_emitterImages[0])->height));
+    _emitterPresets.emplace_back(emitter);
+}
+
+uint32_t ParticleInterface::LoadEmitterImage(const char* imagePath)
```

Comment:
We should really make a global function for loading images... put a TODO

Reply:
I added a card on Codecks for this https://bubonic-brotherhood.codecks.io/decks/7-engine/card/1e1-image-loader-utility-function